search/search-client-query-transformer/src/test/java/io/camunda/search/clients/transformers/filter/ProcessDefinitionQueryTransformerTest.java

```diff
@@ -98,6 +101,37 @@ public void shouldQueryByVersion() {
             });
   }
 
+  @ParameterizedTest
+  @ValueSource(booleans = {true, false})
+  public void shouldQueryByHasStartForm(final boolean hasStartForm) {
+    final var filter = FilterBuilders.processDefinition(f -> f.hasStartForm(hasStartForm));
+
+    // when
+    final var searchRequest = transformQuery(filter);
+
+    // then
+    final var queryVariant = searchRequest.queryOption();
+
+    // then
+    assertThat(searchRequest.queryOption())
+        .isInstanceOfSatisfying(
+            SearchBoolQuery.class,
+            boolQuery -> {
+              final var innerQuery =
+                  hasStartForm
+                      ? boolQuery.must().get(0).queryOption()
+                      : boolQuery.mustNot().get(0).queryOption();
+              final var boolQueryMustOrNot = hasStartForm ? boolQuery.must() : boolQuery.mustNot();
+
+              assertThat(boolQueryMustOrNot).hasSize(1);
```

Comment:
Nitpick: Could this perhaps be refactored a bit?

```suggestion

final var boolQueryMustOrNot = hasStartForm ? boolQuery.must() : boolQuery.mustNot();
assertThat(boolQueryMustOrNot).hasSize(1);
final var innerQuery = boolQueryMustOrNot.get(0).queryOption();

```